sphinx/tutorial/rose/furthertopics/failif-warnif.rst

```diff
@@ -0,0 +1,240 @@
+.. include:: ../../../hyperlinks.rst
+   :start-line: 1
+
+Fail-If, Warn-If
+================
+
+Basic validation can be achieved using metadata settings such as ``type`` and
+``range``. The ``fail-if`` and ``warn-if`` metadata settings are scriptable
+enabling more advanced validation, they evaluate logical expressions,
+flagging warnings if they return false.
+
+``fail-if`` and ``warn-if`` can be run on the command line using ``rose macro``
+or on-demand in the ``rose edit`` GUI.
+
+.. note::
+
+   Simple metadata settings such as ``range`` can be evaluated on-the-fly when
+   a value changes. As ``fail-if`` and ``warn-if`` can take longer to evaluate
+   they must be done on-demand in the ``rose edit`` GUI.
```

Comment:
This seems to me to contradict the previous sentence in the main text (lines 12-13). Do you mean 'they must be done [via one method from either of the two aforementioned]' but while 'on-the-fly' (I assume this means while the suite in question is running) it is only possible to run them via the GUI? I would clarify this more explicitly.